local_pathfinding/node_navigate.py

```diff
@@ -77,14 +78,17 @@ def __init__(self):
             timer_period_sec=pub_period_sec, callback=self.desired_heading_callback
         )
 
+        planner = self.get_parameter("path_planner").get_parameter_value().string_value
+        self.get_logger().debug(f"Got parameter: {planner=}")
```

Comment:
Just realized we can't just read parameter in init because updating during runtime won't work. Can you move getting parameters to its own function. This function would get all parameters, and if they changed then store in instance variable and debug log its new value. Then you can call the function both in the init function and desired heading callback.